src/components/map/LayersService.js

```diff
@@ -775,50 +778,21 @@ goog.require('ga_urlutils_service');
             }
             if (config.sourceId && config.styleUrl) {
               var sourceId = config.sourceId;
-              $http.get(config.styleUrl, {
-                cache: true
-              }).then(function(response) {
-                var glStyle = response.data;
-
-                if (config.sourceType !== 'raster') {
-
-                  // HACK: Make the Swiss style transparent to see the relief
-                  glStyle.layers.forEach(function(style, idx) {
-                    if (style.id === 'Schweiz') {
-                      style.paint['fill-opacity'] = 0.8;
-                    }
-                  });
-
-                  // For vector source we parse then apply the style after
-                  // loading the sprite image.
-                  var spriteConfigUrl = glStyle.sprite + '.json';
-                  var spriteUrl = glStyle.sprite + '.png';
-
-                  // Load the style for a specific source
-                  $http.get(spriteConfigUrl, {
-                    cache: true
-                  }).then(function(response) {
-                    return response.data;
-                  }, function(reason) {
-                    $window.console.error('Loading sprite failed. Reason: "' +
-                        reason.data + '"');
-                    return {};
-                  }).then(function(spriteData) {
-                    $window.olms.stylefunction(olLayer, glStyle,
-                        config.sourceId,
-                        undefined, spriteData, spriteUrl,
-                        ['Helvetica']);
-                  });
-                } else {
-                  glStyle.layers.forEach(function(style, idx) {
-                    if (style.source === config.sourceId) {
-                      // TODO: We should be able to parse the style for raster.
-                      // Verify if it exists or not in mb2olstyle library
-                    }
-                  });
-                }
+              gaGLStyle.get(config.styleUrl).then((data) => {
+                var glStyle = data.styleJSON;
+                var spriteData = data.spriteJSON;
+                var spriteUrl = glStyle.sprite + '.png';
+                $window.olms.stylefunction(olLayer, glStyle,
+                    config.sourceId,
+                    undefined, spriteData, spriteUrl,
+                    ['Helvetica']);
+                // HACK: Make the Swiss style transparent to see the relief
+                glStyle.layers.forEach(function(style, idx) {
```

Comment:
We need to check if we still need this Hack

Reply:
I tried to remove it and didn t see any difference. But provided that we don't really have a swiss use case... I left it just in case.